assets/src/edit-story/components/library/libraryUploadDropTarget.js

```diff
@@ -34,16 +34,41 @@ import {
   UploadDropTargetMessageOverlay,
 } from '../uploadDropTarget';
 import { useUploader } from '../../app/uploader';
+import { useMedia } from '../../app/media';
+import {
+  getResourceFromLocalFile,
+  getResourceFromUploadAPI,
+} from '../../components/library/panes/media/mediaUtils';
 
 const MESSAGE_ID = 'edit-story-library-upload-message';
 
 function LibraryUploadDropTarget({ children }) {
   const { uploadFile } = useUploader();
+  const {
+    state: { media, mediaType, searchTerm },
+    actions: { fetchMediaSuccess },
+  } = useMedia();
   const onDropHandler = useCallback(
-    (files) => {
-      files.forEach(uploadFile);
+    async (files) => {
+      const localFiles = files.map(getResourceFromLocalFile);
+      const localMedia = await Promise.all(localFiles);
+      const filesUploading = files.map((file) => uploadFile(file));
+      fetchMediaSuccess({
+        media: [...localMedia, ...media],
```

Comment:
These are not the same types, are these? This will really hurt us very soon. Let's address right away. We should either use the types that are in the `media: []` reducer, or we should convert them all to `resource` object.

Reply:
For me, those are subtypes of the media/resource. What do you suggest? Update the current media types to support the `local` media? I mean, we should flag than somewhat because we need to consider that the Uploading UI being invoked for local media being uploaded. Not sure if I catch your concern here.

Reply:
These seem to me two completely different object shapes. TBH I'm confused about them myself. My main source of reference is https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/components/library/panes/media/mediaUtils.js. From various fetch APIs we are receiving:

1. "WP Attachment object" with the shape `{src, url, mimeType, oWidth, oHeight, id, ...}`
2. "WP Media Picker object" with the shape `{src, url, mime, width, height, featured_media, featured_media_src, ...}`

Before we construct an element for any of these cases, we convert them to a "resource" object with the shape `{type, src, width, height, mimeType, ...}`.

Not sure which shape are the objects in the `media`, but I think it's the "WP Attachment object". So mixing them with the "resource"-shaped objects is a problem. This could possibly be solved by converting `media` elements to resources right away in our media reducer as soon as we receive them from the fetch API. But I'm not 100% sure of all uses.